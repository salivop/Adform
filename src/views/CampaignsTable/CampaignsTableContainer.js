import React from "react";
import axios from "axios";
import Typography from "@material-ui/core/Typography/Typography";

import PaperWrapper from "components/PaperWrapper/PaperWrapper";
import Loader from "components/Loader/Loader";

import CampaignsTable from "./Table/CampaignsTable";

import { mockTableData } from "./MockData";

class CampaignsTableContainer extends React.Component {
  constructor() {
    super();

    this.state = {
      tableData: [],
      isLoading: true,
      isError: false,
      url: "https://jsonplaceholder.typicode.com/users"
    };
    this.getData = this.getData.bind(this);
    this.dateCheck = this.dateCheck.bind(this);
  }
  componentDidMount() {
    setTimeout(function() {
     this.getData()
  }.bind(this), 2000);
}

  dateCheck = (startDate, endDate) => {
    let currentDate = new Date();
    let cDate = Date.parse(currentDate);
    return cDate <= Date.parse(endDate) && cDate >= Date.parse(startDate)
      ? true
      : false;
  };

  async getData() {
    const { url } = this.state;
    await axios
      .get(url)
      .then(data => {
        this.setState({
          // TODO:
          // THE API IS CHANGED BECAUSE PROVIDED API RETURN WRONG DATA.
          // IN THAT CASE I USED DATA FROM WORD DOC THAT I HAVE RECEIVED FROM ADFORM, ALSO DUE THAT SOME TESTS FAILS
          // tableData: data.data.map(item => ({
          //   ...item,
          //   active: this.dateCheck(item.startDate, item.endDate)
          // })),
          tableData: mockTableData.map(item => ({
            ...item,
            active: this.dateCheck(item.startDate, item.endDate)
          })),
          isLoading: false,
          isError: false
        });
      })
      .catch(this.setState({ isLoading: false, isError: true }));
  }

  render() {
    const { isLoading, isError, tableData } = this.state;
    return (
      <PaperWrapper
        child={
          <React.Fragment>
            {isError && (
              <Typography>Something wrong with table data</Typography>
            )}
            {isLoading ? <Loader /> : <CampaignsTable tableData={tableData} />}
          </React.Fragment>
        }
      />
    );
  }
}

export default CampaignsTableContainer;
