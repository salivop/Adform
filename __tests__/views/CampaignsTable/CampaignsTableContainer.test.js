import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";

import CampaignsTableContainer from "src/views/CampaignsTable/CampaignsTableContainer";

describe("CampaignsTableContainer", () => {
  let wrapper;
  let mock;
  let instance;

  beforeAll(() => {
    mock = new MockAdapter(axios);
  });

  beforeEach(() => {
    mock.reset();
    wrapper = shallow(<CampaignsTableContainer />);
    instance = wrapper.instance();
  });

  it("should render correctly", async () => {
    mock.onGet("https://jsonplaceholder.typicode.com/users").reply(200, []);
    await instance.getData();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should start with an empty initial values", async () => {
    mock.onGet("https://jsonplaceholder.typicode.com/users").reply(200, []);
    await instance.getData();
    expect(wrapper.state().tableData.length).toBe(0);
    expect(wrapper.state().isLoading).toBeFalsy();
    expect(wrapper.state().isError).toBeFalsy();
    expect(wrapper.state().url).toEqual(
      "https://jsonplaceholder.typicode.com/users"
    );
  });

  it("should have PaperWrapper", () => {
    expect(wrapper.find("PaperWrapper").length).toEqual(1);
  });
});