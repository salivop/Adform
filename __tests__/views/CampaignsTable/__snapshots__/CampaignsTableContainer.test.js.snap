// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignsTableContainer should render correctly 1`] = `
<PaperWrapper
  child={
    <React.Fragment>
      <CampaignsTable
        tableData={
          Array [
            Object {
              "Budget": 88377,
              "active": false,
              "endDate": "3/9/2018",
              "id": 1,
              "name": "Divavu",
              "startDate": "9/19/2017",
              "userId": 3,
            },
            Object {
              "Budget": 608715,
              "active": true,
              "endDate": "2/21/2020",
              "id": 2,
              "name": "Jaxspan",
              "startDate": "11/21/2017",
              "userId": 6,
            },
            Object {
              "Budget": 239507,
              "active": false,
              "endDate": "6/20/2017",
              "id": 3,
              "name": "Miboo",
              "startDate": "11/1/2017",
              "userId": 7,
            },
            Object {
              "Budget": 179838,
              "active": false,
              "endDate": "11/30/2017",
              "id": 4,
              "name": "Trilith",
              "startDate": "8/25/2017",
              "userId": 1,
            },
            Object {
              "Budget": 837850,
              "active": false,
              "endDate": "3/10/2018",
              "id": 5,
              "name": "Layo",
              "startDate": "11/28/2017",
              "userId": 9,
            },
            Object {
              "Budget": 858131,
              "active": true,
              "endDate": "6/23/2020",
              "id": 6,
              "name": "Photojam",
              "startDate": "7/25/2017",
              "userId": 3,
            },
            Object {
              "Budget": 109078,
              "active": false,
              "endDate": "1/15/2018",
              "id": 7,
              "name": "Blogtag",
              "startDate": "6/27/2017",
              "userId": 2,
            },
            Object {
              "Budget": 272552,
              "active": false,
              "endDate": "1/25/2018",
              "id": 8,
              "name": "Rhyzio",
              "startDate": "10/13/2017",
              "userId": 4,
            },
            Object {
              "Budget": 301919,
              "active": false,
              "endDate": "11/10/2017",
              "id": 9,
              "name": "Zoomcast",
              "startDate": "9/6/2017",
              "userId": 8,
            },
            Object {
              "Budget": 505602,
              "active": false,
              "endDate": "10/2/2017",
              "id": 10,
              "name": "Realbridge",
              "startDate": "3/5/2018",
              "userId": 5,
            },
          ]
        }
      />
    </React.Fragment>
  }
/>
`;
